[Syntax Compatibility] Column aliases do not support double quotes

The error message is as follows, unable to add double quotes to the alias

```
panic: line 0:17 expected the last token kind is: <ident>, but got <string>
SELECT 'abc' as "value2"
                 ^^^^^^
```

--- a/parser/ast.go
+++ b/parser/ast.go
@@ -3946,7 +3946,7 @@ func (n *IsNotNullExpr) Accept(visitor ASTVisitor) error {
 type AliasExpr struct {
 	Expr     Expr
 	AliasPos Pos
-	Alias    *Ident
+	Alias    Expr
 }
 
 func (a *AliasExpr) Pos() Pos {
@@ -3954,7 +3954,7 @@ func (a *AliasExpr) Pos() Pos {
 }
 
 func (a *AliasExpr) End() Pos {
-	return a.Alias.NameEnd
+	return a.Alias.End()
 }
 
 func (a *AliasExpr) String(level int) string {

--- a/parser/parser_column.go
+++ b/parser/parser_column.go
@@ -21,14 +21,14 @@ func (p *Parser) parseExpr(pos Pos) (Expr, error) {
 	case p.matchKeyword(KeywordAs): // syntax: columnExpr (alias | AS identifier)
 		aliasPos := p.Pos()
 		_ = p.lexer.consumeToken()
-		asIdent, err := p.parseIdent()
+		alias, err := p.parseIdentOrString(p.Pos())
 		if err != nil {
 			return nil, err
 		}
 		return &AliasExpr{
 			AliasPos: aliasPos,
 			Expr:     orExpr,
-			Alias:    asIdent,
+			Alias:    alias,
 		}, nil
 	}
 	return orExpr, nil

--- a/parser/parser_common.go
+++ b/parser/parser_common.go
@@ -92,6 +92,17 @@ func (p *Parser) parseIdent() (*Ident, error) {
 	return ident, nil
 }
 
+func (p *Parser) parseIdentOrString(_ Pos) (Expr, error) {
+	switch {
+	case p.matchTokenKind(TokenIdent):
+		return p.parseIdent()
+	case p.matchTokenKind(TokenString):
+		return p.parseString(p.Pos())
+	default:
+		return nil, fmt.Errorf("expected <ident> or <string>, but got %q", p.lastTokenKind())
+	}
+}
+
 func (p *Parser) parseIdentOrStar() (*Ident, error) {
 	switch {
 	case p.matchTokenKind(TokenIdent):

--- a/parser/parser_table.go
+++ b/parser/parser_table.go
@@ -539,16 +539,7 @@ func (p *Parser) tryParseOnCluster(pos Pos) (*OnClusterExpr, error) {
 		return nil, err
 	}
 
-	var expr Expr
-	var err error
-	switch {
-	case p.matchTokenKind(TokenIdent):
-		expr, err = p.parseIdent()
-	case p.matchTokenKind(TokenString):
-		expr, err = p.parseString(p.Pos())
-	default:
-		return nil, fmt.Errorf("expected <Ident> or <Literal>, but got %q", p.last().String)
-	}
+	expr, err := p.parseIdentOrString(p.Pos())
 	if err != nil {
 		return nil, err
 	}

--- a/parser/testdata/query/format/select_column_alias_string.sql
+++ b/parser/testdata/query/format/select_column_alias_string.sql
@@ -1,0 +1,7 @@
+-- Origin SQL:
+SELECT 'abc' as "value2";
+
+-- Format SQL:
+
+SELECT 
+  'abc' AS 'value2';

--- a/parser/testdata/query/output/select_column_alias_string.sql.golden.json
+++ b/parser/testdata/query/output/select_column_alias_string.sql.golden.json
@@ -1,0 +1,43 @@
+[
+  {
+    "SelectPos": 0,
+    "StatementEnd": 23,
+    "With": null,
+    "Top": null,
+    "SelectColumns": {
+      "ListPos": 8,
+      "ListEnd": 23,
+      "HasDistinct": false,
+      "Items": [
+        {
+          "Expr": {
+            "LiteralPos": 8,
+            "LiteralEnd": 11,
+            "Literal": "abc"
+          },
+          "AliasPos": 13,
+          "Alias": {
+            "LiteralPos": 17,
+            "LiteralEnd": 23,
+            "Literal": "value2"
+          }
+        }
+      ]
+    },
+    "From": null,
+    "ArrayJoin": null,
+    "Window": null,
+    "Prewhere": null,
+    "Where": null,
+    "GroupBy": null,
+    "WithTotal": false,
+    "Having": null,
+    "OrderBy": null,
+    "LimitBy": null,
+    "Limit": null,
+    "Settings": null,
+    "UnionAll": null,
+    "UnionDistinct": null,
+    "Except": null
+  }
+]

--- a/parser/testdata/query/select_column_alias_string.sql
+++ b/parser/testdata/query/select_column_alias_string.sql
@@ -1,0 +1,1 @@
+SELECT 'abc' as "value2";

